### ISSUE TITLE
Support for Jersey MappableException in 'DefaultJerseyTagsProvider'

### ISSUE THREAD

**Issue by jvalloire** [micrometer-metrics/micrometer#658]
I am using Micrometer in my Jersey application and i have noticed that the `DefaultJerseyTagsProvider` can't handle `MappableException`.

Here is an extract from the `DefaultJerseyTagsProvider` code:
```
 private static int statusCode(RequestEvent event) {
        Throwable exception = event.getException();
        if (exception != null) {
            if (exception instanceof WebApplicationException) {
                return ((WebApplicationException) exception).getResponse().getStatus();
            }
            return Response.Status.INTERNAL_SERVER_ERROR.getStatusCode();
        }
        return event.getContainerResponse().getStatus();
    }
```

With that code, whenever we have an exception that is not a `WebApplicationException`, we will have a **500** status code...

When using `MappableException`, application exceptions are mapped to Jersey Response (with an appropriate response status).
For example, the `ValidationExceptionMapper` from Jersey should return a **400** (_Bad Request_) when request parameters validation fails.
As a consequence, when using **micrometer-jersey2**, i am expecting a `http.server.requests` metric with a _{status = 400}_ tag when my request is not valid but i only have _{status = 500}_ tags.

Should it be possible to add support for `MappableException` so as the status tag value would be the same than the response status code?

**Comment by jvalloire** [comment]
Yes, i would be glad to help.
Do you have any idea of the way `MappableException`s can be handled in the `DefaultJerseyTagsProvider` ?

**Comment by mweirauch** [comment]
As I said on Slack, I can work on this. We just need to coordinate in case you are having a go at it.

The challenge is extracting a useful statuscode in this case. As we are post (or at least optionally post) `EXCEPTION_MAPPING_FINISHED` we can go by checking if the exception was successfully mapped.

@jvalloire Do you have a sample Jersey resource exposing the described problem? (Alternatively I think what is required for replicating is throwing a custom exception for which an `ExceptionMapper` is registered. So we can get passed a `MappableException` and can go on extracting the exceptionmapper-mapped statuscode from the response in `DefaultJerseyTagsProvider`.)

**Comment by jvalloire** [comment]
@mweirauch i'm working on it now.

I add a use-case in `TestResource` test class with an endpoint throwing a custom exception with a `ExceptionMapper` registered for it.

I checked the `RequestEvent` type and since it is `FINISHED`, i think we could simply use the container response status in case of `MappableException` (with a _null-check_ on `event.getContainerResponse()` that might be null if no mapper is found, like in the use case with the `IllegalArgumentException`).

Are you ok with such an implementation ?

**Comment by mweirauch** [comment]
Yeah, also looked a bit and came to the same conclusion. So within the `statusCode()` we don't even have to handle any exceptions, just figure if the `ContainerResponse` is there and use its `getStatus()`, otherwise return `Response.Status.INTERNAL_SERVER_ERROR.getStatusCode()`.

Thanks for looking into it!

PS: PR against branch `1.0.x` please.
